### ISSUE TITLE
about  reproduce

### ISSUE THREAD

**Issue by menghuaa** [piddnad/DDColor#6]
1. The configuration you provided is 4 cards, with each card having a batch size of 4. Have you tried training the model with two cards and each card having a batch size of 8? How would this affect the model's performance?
2. How do you choose a model? Do you directly select the model that has trained 400000 steps as the selected model?
3. May I ask if the training code you provided is DDColor-large?

**Comment by piddnad** [comment]
Thanks for your interest in our work. 
1. We haven't tried the training settings you mentioned, but I think if the batch size remains the same, the resulting performance should not change much. 
2. For the paper model, we simply choose the model with 400,000 steps. In practice, you can also choose the model with the best performance (i.e. the lowest FID) refering to the training log. 
3. Yes. But you can train the smaller model by simply modifying the "encoder_name" in options/train/train_ddcolor.yml.

**Comment by menghuaa** [comment]
> Thanks for your interest in our work.
> 
> 1. We haven't tried the training settings you mentioned, but I think if the batch size remains the same, the resulting performance should not change much.
> 2. For the paper model, we simply choose the model with 400,000 steps. In practice, you can also choose the model with the best performance (i.e. the lowest FID) refering to the training log.
> 3. Yes. But you can train the smaller model by simply modifying the "encoder_name" in options/train/train_ddcolor.yml.
I just changed the GPU to 2, and the batch size of each GPU to 8. The other parameters are not modified.  The results I reproduced were tested on the same test set as the ddcolor-paper weight you provided, and the results are as follows. Is this fid gap within the normal range?
![image](https://github.com/piddnad/DDColor/assets/48876313/5147102f-a718-4c82-b1b4-332e609c2ddc)

**Comment by piddnad** [comment]
> 3\. I just changed the GPU to 2, and the batch size of each GPU to 8. The other parameters are not modified.  The results I reproduced were tested on the same test set as the ddcolor-paper weight you provided, and the results are as follows. Is this fid gap within the normal range?

I think that's a reasonable range.

**Comment by menghuaa** [comment]
> > 3. I just changed the GPU to 2, and the batch size of each GPU to 8. The other parameters are not modified.  The results I reproduced were tested on the same test set as the ddcolor-paper weight you provided, and the results are as follows. Is this fid gap within the normal range?
> 
> I think that's a reasonable range.
The following test results are based on the model with the lowest fid I have selected and the results of the ddcolor-paper weight test you provided. There is a significant difference in the value of 'f'. Is this normal?
![image](https://github.com/piddnad/DDColor/assets/48876313/0f5c8ffb-49ad-4bbe-9e51-785a8099ed52)

**Comment by piddnad** [comment]
> > > 3. I just changed the GPU to 2, and the batch size of each GPU to 8. The other parameters are not modified.  The results I reproduced were tested on the same test set as the ddcolor-paper weight you provided, and the results are as follows. Is this fid gap within the normal range?
> > 
> > 
> > I think that's a reasonable range.
> > The following test results are based on the model with the lowest fid I have selected and the results of the ddcolor-paper weight test you provided. There is a significant difference in the value of 'cf'. Is this normal?
> > ![image](https://github.com/piddnad/DDColor/assets/48876313/0f5c8ffb-49ad-4bbe-9e51-785a8099ed52)

I don't think it's normal, the difference in cf is a bit too large.

**Comment by menghuaa** [comment]
> > > > 3. I just changed the GPU to 2, and the batch size of each GPU to 8. The other parameters are not modified.  The results I reproduced were tested on the same test set as the ddcolor-paper weight you provided, and the results are as follows. Is this fid gap within the normal range?
> > > 
> > > 
> > > I think that's a reasonable range.
> > > The following test results are based on the model with the lowest fid I have selected and the results of the ddcolor-paper weight test you provided. There is a significant difference in the value of 'cf'. Is this normal?
> > > ![image](https://github.com/piddnad/DDColor/assets/48876313/0f5c8ffb-49ad-4bbe-9e51-785a8099ed52)
> 
> I don't think it's normal, the difference in cf is a bit too large.

Have you ever used the code you provided for reproduce?

**Comment by piddnad** [comment]
> > > > > 3. I just changed the GPU to 2, and the batch size of each GPU to 8. The other parameters are not modified.  The results I reproduced were tested on the same test set as the ddcolor-paper weight you provided, and the results are as follows. Is this fid gap within the normal range?
> > > > 
> > > > 
> > > > I think that's a reasonable range.
> > > > The following test results are based on the model with the lowest fid I have selected and the results of the ddcolor-paper weight test you provided. There is a significant difference in the value of 'cf'. Is this normal?
> > > > ![image](https://github.com/piddnad/DDColor/assets/48876313/0f5c8ffb-49ad-4bbe-9e51-785a8099ed52)
> > 
> > 
> > I don't think it's normal, the difference in cf is a bit too large.
> 
> Have you ever used the code you provided for reproduce?

The models we provide are trained based on the provided code. 
You can check your specific environment and configuration, but please note that different environments and machines do lead to slightly different test results. 
If you still have the problem of too large CF value when training your own model, you can try to reduce the weight of CL during training.

**Comment by piddnad** [comment]
Thank you for your feedback. The environment has been indicated in README.md. There is really no fixed standard range for the indicators fid, cf and psnr. The value range of these indicators can vary depending on the dataset and task. Therefore, a universal standard range cannot be provided. Instead, we can evaluate and analyze by comparing the results in the same experiment or the literature of similar studies in the reference field.

**Comment by menghuaa** [comment]
> 感谢您的反馈意见。环境已在README.md中标明。fid、cf和psnr指标确实没有固定的标准范围。这些指标的值范围可能会根据数据集和任务的不同而有所不同。因此，无法提供通用的标准范围。相反，我们可以通过比较同一实验的结果或参考领域类似研究的文献来评估和分析。

May I ask if you trained on the ImageNet_2012/ILSVRC2012_train dataset? It contains 1281167 images.

**Comment by piddnad** [comment]
Yes. Please refer to paper sec 4.1 for more information.

**Comment by menghuaa** [comment]
> Yes. Please refer to paper sec 4.1 for more information.

Could you please provide a training log？
